homeassistant/components/dlink/switch.py

```diff
@@ -84,52 +83,28 @@ async def async_setup_entry(
 ) -> None:
     """Set up the D-Link Power Plug switch."""
     async_add_entities(
-        [
-            SmartPlugSwitch(
-                hass,
-                entry,
-                hass.data[DOMAIN][entry.entry_id],
-                SWITCH_TYPE,
-            ),
-        ],
+        [SmartPlugSwitch(entry, hass.data[DOMAIN][entry.entry_id], SWITCH_TYPE)],
         True,
     )
 
 
 class SmartPlugSwitch(DLinkEntity, SwitchEntity):
     """Representation of a D-Link Smart Plug switch."""
 
-    def __init__(
-        self,
-        hass: HomeAssistant,
-        entry: ConfigEntry,
-        data: SmartPlugData,
-        description: SwitchEntityDescription,
-    ) -> None:
-        """Initialize the switch."""
-        super().__init__(data, entry, description)
-        self.units = hass.config.units
-
     @property
     def extra_state_attributes(self) -> dict[str, Any]:
         """Return the state attributes of the device."""
-        try:
-            ui_temp = self.units.temperature(
-                int(self.data.temperature or 0), UnitOfTemperature.CELSIUS
+        attrs: dict[str, Any] = {}
+        if self.data.temperature:
+            attrs[ATTR_TEMPERATURE] = self.hass.config.units.temperature(
+                int(self.data.temperature), UnitOfTemperature.CELSIUS
             )
-            temperature = ui_temp
-        except (ValueError, TypeError):
```

Comment:
Are we sure that it's safe to remove the error catching?

Reply:
TypeError won't happen now thanks to mypy. Just fixed the ValueError.

Reply:
Should we check for temperature too?